test/note_test.rb

```diff
@@ -2,19 +2,25 @@
 
 class TestNote < MiniTest::Test
   def setup
-    @frequency   = 220.to_f
-    @duration    = 2.0
-    @sample_rate = 22050
-    @note = MusicTheory::Note.new(frequency: @frequency, duration: @duration)
+    @test_samples = eval(File.read('test/fixtures/note_samples_array'))
+    @basic_note = MusicTheory::Note.new
+    @note_220 = MusicTheory::Note.new(frequency: 220)
+    @distorted_note = MusicTheory::Note.new(distort: true)
   end
 
-  def test_total_frames
-    frames = (@duration * @sample_rate).to_i
-    assert_equal frames, @note.total_frames
+  def test_samples_are_identical
+    assert_equal @test_samples, @basic_note.samples
```

Comment:
You could use a [hash function](http://ruby-doc.org/stdlib-2.2.0/libdoc/digest/rdoc/Digest.html) here as a kind of [checksum](http://en.wikipedia.org/wiki/Checksum) to still ensure that the two are identical, with the added benefit that if/when this test fails, it'll print the diff of the hashed values instead of the massive samples array. Something like
```ruby
require 'digest'
...
digest = Digest::MD5.new
assert_equal digest.base64digest(@test_samples), digest.base64digest(@basic_note.samples)
...
```
Hash functions are worth reading up on as they turn up a lot. A few example use-cases are cache invalidation (which is similar to what I'm proposing here), cryptography ([RSA](http://en.wikipedia.org/wiki/RSA_%28cryptosystem%29), for example), and of course implementing hash tables (dictionaries, or just the 'Hash' data structure in Ruby).